### PR TITLE
Make default rowset type to config

### PR DESCRIPTION
**Pull request by kangpinghuang**
For test convenience and rolling deploy, the default rowset type should be configurable.
I use two configs because I will control loading and compaction seperately. BTW, schema change will use the original rowset type of the base table.

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -481,6 +481,12 @@ namespace config {
     // config for tablet meta checkpoint
     CONF_Int32(tablet_meta_checkpoint_min_new_rowsets_num, "10");
     CONF_Int32(tablet_meta_checkpoint_min_interval_secs, "600");
+
+    // config for default rowset type
+    // 0: ALPHA_ROWSET
+    // 1: BETA_ROWSET
+    CONF_Int32(default_rowset_type, "0");
+    CONF_Int32(compaction_rowset_type, "0");
```
Comment (imay) [inline be/src/common/config.h]: It will be better to use string, and we convert string to enum with code. It will make sense.

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/common/config.h (hunk 1 above)
Comment (kangpinghuang) [reply be/src/common/config.h]: ok

File: be/src/olap/storage_engine.cpp [hunk 2]
```diff
@@ -733,6 +733,25 @@ OLAPStatus StorageEngine::_do_sweep(
     return res;
 }
 
+// invalid rowset type config will return ALPHA_ROWSET for system to run smoothly
+void StorageEngine::_parse_default_rowset_type() {
```
Comment (imay) [inline be/src/olap/storage_engine.cpp]: No one call this function?

File: be/src/olap/storage_engine.h [hunk 3]
```diff
@@ -357,6 +364,13 @@ class StorageEngine {
 
     MemTableFlushExecutor* _memtable_flush_executor;
 
+    // default rowset type for load
+    // used to decide the type of new loaded data
+    RowsetTypePB _default_rowset_type;
```
Comment (imay) [inline be/src/olap/storage_engine.h]: no default value?

File: be/src/olap/storage_engine.h [hunk 4]
```diff
@@ -206,6 +206,10 @@ class StorageEngine {
 
     MemTableFlushExecutor* memtable_flush_executor() { return _memtable_flush_executor; }
 
+    RowsetTypePB default_rowset_type() { return _default_rowset_type; }
```
Comment (imay) [inline be/src/olap/storage_engine.h]:
```suggestion
    RowsetTypePB default_rowset_type() const { return _default_rowset_type; }
```

File: be/src/common/config.h [hunk 5]
```diff
@@ -481,6 +481,11 @@ namespace config {
     // config for tablet meta checkpoint
     CONF_Int32(tablet_meta_checkpoint_min_new_rowsets_num, "10");
     CONF_Int32(tablet_meta_checkpoint_min_interval_secs, "600");
+
+    // config for default rowset type
+    // Valid configs: ALPHA_ROWSET, BETA_ROWSET
+    CONF_String(default_rowset_type, "ALPHA_ROWSET");
```
Comment (imay) [inline be/src/common/config.h]: I think "ALPHA", "BETA" is enough

File: be/src/olap/storage_engine.h (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/storage_engine.h]: ok

File: be/src/olap/storage_engine.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/storage_engine.cpp]: miss it. I will add it to open()

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**